tape/src/main/java/com/squareup/tape/ObjectQueue.java

```diff
@@ -64,19 +78,32 @@ void clear() throws IOException {
    * already in the queue. If an error occurs while reading the data, the listener will not receive
    * further notifications.
    */
-  abstract void setListener(Listener<T> listener) throws IOException;
+  public abstract void setListener(Listener<T> listener) throws IOException;
 
   /**
    * Listens for changes to the queue.
    *
    * @param <T> The type of elements in the queue.
    */
-  interface Listener<T> {
+  public interface Listener<T> {
 
     /** Called after an entry is added. */
     void onAdd(ObjectQueue<T> queue, T entry);
 
     /** Called after an entry is removed. */
     void onRemove(ObjectQueue<T> queue);
   }
+
+  /**
+   * Convert a byte stream to and from a concrete type.
+   *
+   * @param <T> Object type.
+   */
+  public interface Converter<T> {
```

Comment:
Oh yeah it's awful naming. Didn't want to change it here.

On Mon, Sep 5, 2016, 10:28 PM Jesse Wilson <notifications@github.com> wrote:

> In tape/src/main/java/com/squareup/tape/ObjectQueue.java
> <https://github.com/square/tape/pull/113#discussion_r77565942>:
>
> >
> >      /** Called after an entry is added. */
> >      void onAdd(ObjectQueue<T> queue, T entry);
> >
> >      /** Called after an entry is removed. */
> >      void onRemove(ObjectQueue<T> queue);
> >    }
> > +
> > +  /**
> > +   * Convert a byte stream to and from a concrete type.
> > +   *
> > +   * @param <T> Object type.
> > +   */
> > +  public interface Converter<T> {
>
> Though fine for now!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/square/tape/pull/113/files/1ff8646d91f328ea567d54874474a7c1f812a010#r77565942>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEEEfw78HRZn-YudoFC7KL2_-zDQAx7ks5qnM_bgaJpZM4J1YlZ>
> .
>